app/models/workspace.rb

```diff
@@ -2,4 +2,9 @@
 
 class Workspace < ApplicationRecord
   belongs_to :project
+
+  def embedded_workspace_state
```

Comment:
This could be an attribute / option of the workspace I guess? Or maybe we don't want to yet give that flexibility